packages/react-popup/Spec.md

```diff
@@ -1,31 +1,385 @@
+- Inline popup rendered in DOM order
+- Open on:
+  - Click
+  - Hover
+  - Context menu (right click)
+  - Focus
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+The `Popover` component will use React context to manage both a trigger and content component.
+
+### Popover
+
+Outer component that setups context and does not render DOM.
+
+> TODO Discuss: dismiss on scroll ? Seems to be a thing in v8 because callout doesn't support repositioning
+
+> TODO Discuss: v8 `hidden` `shouldUpdateHidden` prop [#4419](https://github.com/microsoft/fluentui/pull/4419) [#10465](https://github.com/microsoft/fluentui/pull/10465)
+
+> TODO Discuss: v8 `onPositioned`
+
+> TODO Discuss: v0/v8 `autoFocus/setInitialFocus` ? -> ***(not focus trap)** can be achieved pretty easily in userland with/without Tabster, in Teams this is extremely rare
+
+> TODO Discuss: A11y -> Should only one popup be open at a time or is `aria-hidden` enough ?
+
+> TODO Discuss: merge `position` and `align` props -> no real reason they were separated in v0 in the first place
+
+> TODO Discuss: start small with API base for positioning props ?
+
+The `@fluentui/react-positioning` library that exports the `usePopper` hook which will power the `Popover` contains more than the declared props here. These extra positioning props should be exposed as required.
+
+```typescript
+export interface PopoverProps {
+  /**
+   * Controls the popup open state
+   */
+  open?: boolean;
+
+  /**
+   * Call back when the component requests to change value
+   */
+  onOpenChange?: (e: OpenPopoverEvents, data: OpenEventData) => void;
+
+  /**
+   * Anchor the popup to an element other than the trigger
+   */
+  target?: HTMLElement;
+
+  /**
+   * Popover position relative to target
+   */
+  position?: string;
+
+  /**
+   * Popover alignment relative to target
+   */
+  align?: string;
+
+  /**
+   * Popover offset value or callback with positioning props
+   */
+  offset?: OffsetFunction | [number, number];
+
+  /**
+   * Renders `PopoverContent` to a portal out of DOM order
+   *
+   * @defaultValue document.body
+   */
+  mountNode?: string;
+  
+  /**
+   * Explicitly render the popup in DOM order
+   */
+  inline?: boolean;
+  
+  /**
+   * Traps focus inside the popup and applies modal dialog behaviour
+   */
+  trapFocus?: boolean;
+}
+```
+
+### PopoverTrigger
+
+This component does not render DOM. Utility component that clones a single child and applies HTML event callbacks to control the open/dismiss of the popup.
+
+```typescript
+export interface PopoverTriggerProps {
+  /**
+   * Should only be a single child
+   */
+  children?: React.ReactElement;
+}
+```
+
+### PopoverContent
+
+This component renders the positioned HTML element and renders user provided children. Renders as `<div>` by default.
+
+```typescript
+export interface PopoverTriggerProps {
+  children?: React.ReactNode;
+}
```

Comment:
Should this be the following?

```suggestion
export interface PopoverContentProps {
  children?: React.ReactNode[];
}
```

Reply:
yeah the name is a good catch but no need for array, `ReactNode` is essentially `anything`

Reply:
changed to `React.ReactElement`